lib/helper/Testcafe.js

```diff
@@ -0,0 +1,661 @@
+// @ts-nocheck
+const fs = require('fs');
+const assert = require('assert');
+const path = require('path');
+const requireg = require('requireg');
+const createTestCafe = require('testcafe');
+const { Selector, ClientFunction } = require('testcafe');
+
+const testControllerHolder = require('./testcafe/testControllerHolder');
+const { mapError, createTestFile } = require('./testcafe/testcafe-utils');
+
+const stringIncludes = require('../assert/include').includes;
+const { urlEquals } = require('../assert/equal');
+const {
+  xpathLocator,
+} = require('../utils');
+const Locator = require('../locator');
+const Helper = require('../helper');
+
+/**
+ * Client Functions
+ */
+const getPageUrl = t => ClientFunction(() => document.location.href).with({ boundTestRun: t });
+
+/**
+ * Uses Testcafe to run tests.
+ */
+class Testcafe extends Helper {
+  constructor(config) {
+    super(config);
+
+    this.iteration = 1;
+    this.testcafe = undefined; // testcafe instance
+    this.t = undefined; // testcafe test controller
+    this.dummyTestcafeFile; // generated testcafe test file
+
+    this.context = undefined; // TODO Not sure if this applies to testcafe
```

Comment:
context is used for `within()` function.
It requires to have `_withinBegin`and `_withinEnd` implemented. 
Inside `_withinBegin` we should define that all next element calls should be started from a specific element (this.context).

Reply:
ok